fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/tree/TreeTypeHelper.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.tree;
+
+import com.mojang.serialization.Codec;
+import net.fabricmc.fabric.mixin.object.builder.BlockStateProviderTypeInvoker;
+import net.fabricmc.fabric.mixin.object.builder.FoliagePlacerTypeInvoker;
+import net.fabricmc.fabric.mixin.object.builder.TreeDecoratorTypeInvoker;
+import net.fabricmc.fabric.mixin.object.builder.TrunkPlacerTypeInvoker;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.intprovider.IntProvider;
+import net.minecraft.util.math.intprovider.IntProviderType;
+import net.minecraft.world.gen.foliage.FoliagePlacer;
+import net.minecraft.world.gen.foliage.FoliagePlacerType;
+import net.minecraft.world.gen.stateprovider.BlockStateProvider;
+import net.minecraft.world.gen.stateprovider.BlockStateProviderType;
+import net.minecraft.world.gen.treedecorator.TreeDecorator;
+import net.minecraft.world.gen.treedecorator.TreeDecoratorType;
+import net.minecraft.world.gen.trunk.TrunkPlacer;
+import net.minecraft.world.gen.trunk.TrunkPlacerType;
+
+/**
+ * A registry for {@link TrunkPlacerType}s, {@link FoliagePlacerType}s, {@link TreeDecoratorType}s,
+ * {@link BlockStateProviderType}s, and {@link IntProviderType}s.
+ */
+public final class TreeTypeHelper {
+	/**
+	 * Creates a new instance of a {@link TrunkPlacerType}, registers and returns it.<br>
+	 * Uses {@link TrunkPlacerTypeInvoker} under the hood.
```

Comment:
```suggestion
	 * Creates a new instance of a {@link TrunkPlacerType}, registers and returns it.
```
No need to mandate implementation details.